### DISCUSSION TITLE
Change default extensions location for remote development in VS code

### DISCUSSION THREAD

**Discussion by JheffSan** [microsoft/vscode-discussions#856] [Extension Development QnA] [answered]
My Visual Studio Code was installed on my main driver - Local Disk (C:) and has also installed the remote ssh extension. However, local disk c is already full and there's no space anymore. I can't connect to the host server anymore because the error said there's no space on my local disk c.

Please see this link for the image: https://snipboard.io/EHNWTP.jpg

I tried to put the path of the config file in another disk to remote.ssh: Config File but I think it's not enough. How can I change the location of that extension to another disk especially when connecting to the new host? Is it possible even if the visual studio code software is installed on local c but all the extension and other resources of host/server is on another disk or fileserver? How can I achieve that?

**Comment by tamuratak** [comment, marked as answer]
See https://code.visualstudio.com/docs/editor/command-line#_advanced-cli-options

Advanced CLI options

Argument | Description
-- | --
--extensions-dir dir | Set the root path for extensions. Has no effect in Portable Mode.
--user-data-dir dir | Specifies the directory that user data is kept in, useful when running as root. Has no effect in Portable Mode.

---

You can also use VS Code in Portable Mode: https://code.visualstudio.com/docs/editor/portable

**Comment by MrMarvel** [comment]
What about not cli options?
I use vscode as system app for all users.
And I want to set fixed path like %appdata%/.vscode for every user instead of %user%/.vscode.
I do not want to change shortcuts to add cli arguments for every user in system. 
Windows 11 25H2
